compiler/rustc_target/src/spec/aarch64_nintendo_switch.rs

```diff
@@ -0,0 +1,36 @@
+use super::{LinkerFlavor, LldFlavor, PanicStrategy, RelroLevel, Target, TargetOptions};
+
+const LINKER_SCRIPT: &str = include_str!("./aarch64_nintendo_switch_linker_script.ld");
+
+/// A base target for Nintendo Switch devices using a pure LLVM toolchain.
+pub fn target() -> Target {
+    let mut opts = TargetOptions {
+        linker_flavor: LinkerFlavor::Lld(LldFlavor::Ld),
+        linker: Some("rust-lld".to_owned()),
+        link_script: Some(LINKER_SCRIPT.to_string()),
+        os: "horizon".to_string(),
+        max_atomic_width: Some(128),
+        panic_strategy: PanicStrategy::Abort,
+        position_independent_executables: true,
+        crt_static_default: false,
+        crt_static_respected: false,
+        dynamic_linking: true,
+        executables: true,
+        has_elf_tls: false,
+        has_rpath: false,
+        relro_level: RelroLevel::Off,
+        ..Default::default()
+    };
+
+    opts.pre_link_args.insert(LinkerFlavor::Lld(LldFlavor::Ld), vec![]);
+
+    opts.post_link_args.insert(LinkerFlavor::Lld(LldFlavor::Ld), vec![]);
```

Comment:
A number of fields here use values that are already used by default (see `impl Default for TargetOptions`), they can all be removed.